src/platform_impl/macos/view.rs

```diff
@@ -504,31 +646,25 @@ extern "C" fn insert_text(this: &Object, _sel: Sel, string: id, _replacement_ran
         let state_ptr: *mut c_void = *this.get_ivar("winitState");
         let state = &mut *(state_ptr as *mut ViewState);
 
-        let has_attr: BOOL = msg_send![string, isKindOfClass: class!(NSAttributedString)];
-        let characters = if has_attr != NO {
-            // This is a *mut NSAttributedString
-            msg_send![string, string]
+        let string = id_to_string(string);
+
+        let is_control = if let Some(ch) = string.chars().next() {
+            ch.is_control()
         } else {
-            // This is already a *mut NSString
-            string
+            false
         };
 
-        let slice =
-            slice::from_raw_parts(characters.UTF8String() as *const c_uchar, characters.len());
-        let string = str::from_utf8_unchecked(slice);
-
         // We don't need this now, but it's here if that changes.
         //let event: id = msg_send![NSApp(), currentEvent];
 
-        let mut events = VecDeque::with_capacity(characters.len());
-        for character in string.chars().filter(|c| !is_corporate_character(*c)) {
-            events.push_back(EventWrapper::StaticEvent(Event::WindowEvent {
-                window_id: WindowId(get_window_id(state.ns_window)),
-                event: WindowEvent::ReceivedCharacter(character),
-            }));
+        // Marked text will be clear when inserting marked text.
+        if state.is_ime_enabled() && !is_control {
+            state.ime_state = ImeState::WillCommit { text: string };
```

Comment:
Maybe just return and don't nest `for` loop?

Reply:
I don't know what you are referring to. Please explain

Reply:
I mean, that you can early return, and you won't need else branch. So you won't nest for loop increasing  overall indention.